File: src/tests/MappingsVB/Second.vb
Imports System
Imports System.Data
Imports voidsoft.DataBlock

Namespace Extender
 


		Public Class Second
		             Inherits TableMetadata
			Private _fields() as DatabaseField

			Sub New() 
			
					_fields = new DatabaseField(4) {}
 _fields(0) = new DatabaseField(DbType.Int32,"Id",True,True,Nothing)
 _fields(1) = new DatabaseField(DbType.Int32,"FirstId",False,False,Nothing)
 _fields(2) = new DatabaseField(DbType.Int32,"Age",False,False,Nothing)
 _fields(3) = new DatabaseField(DbType.Int32,"Quantity",False,False,Nothing)
 
me.currentTableName = "Second"

Me.listRelations.Add(new ParentTableRelation("Third", "SecondId",TableRelationCardinality.OneToMany,False))
Me.listRelations.Add(new ChildTableRelation("First",TableRelationCardinality.OneToOne,"Id","FirstId"))

End Sub



			Public Overrides Property TableFields() as DatabaseField() 
		   	   Get
				  Return _fields
			   End Get
			   Set (ByVal Value as DatabaseField())
			      _fields = value
			   End Set
			End Property

     Public Property Id As System.Int32
        Get
           Return  CType((Me.GetField("Id")).fieldValue,System.Int32)
        End Get 
        Set(ByVal Value As System.Int32)
          Me.SetFieldValue(Id, Value)
        End Set
      End Property

     Public Property FirstId As System.Int32
        Get
           Return  CType((Me.GetField("FirstId")).fieldValue,System.Int32)
        End Get 
        Set(ByVal Value As System.Int32)
          Me.SetFieldValue(FirstId, Value)
        End Set
      End Property

     Public Property Age As System.Int32
        Get
           Return  CType((Me.GetField("Age")).fieldValue,System.Int32)
        End Get 
        Set(ByVal Value As System.Int32)
          Me.SetFieldValue(Age, Value)
        End Set
      End Property

     Public Property Quantity As System.Int32
        Get
           Return  CType((Me.GetField("Quantity")).fieldValue,System.Int32)
        End Get 
        Set(ByVal Value As System.Int32)
          Me.SetFieldValue(Quantity, Value)
        End Set
      End Property
Public Function GetThird() As Array
    Dim relatedType As Type = Type.GetType("Extender.Third")
    Return Me.GetRelatedTableData(relatedType)
End Function
Public Function GetFirst() As TableMetadata
    Dim relatedType As Type = Type.GetType("Extender.First")
    Dim result() As TableMetadata = Me.GetRelatedTableData(relatedType)
   If result.Length > 0 Then
      Return result(0)
   Else
      Return Nothing
   End If
End Function
End Class
End Namespace

File: src/tests/Mappings/Second.cs
using System;
using System.Data;
using voidsoft.DataBlock;

namespace Extender
{

      [Serializable()]
		public class Second : TableMetadata
		{

                   public enum SecondFields
                   {
                      Id,
                      FirstId,
                      Age,
                      Quantity
                  }


			    private DatabaseField[] _fields;

		    	public Second()
			    {
					    _fields = new DatabaseField[4];
                    _fields[0] = new DatabaseField(DbType.Int32,"Id",true,true,null);
                    _fields[1] = new DatabaseField(DbType.Int32,"FirstId",false,false,null);
                    _fields[2] = new DatabaseField(DbType.Int32,"Age",false,false,null);
                    _fields[3] = new DatabaseField(DbType.Int32,"Quantity",false,false,null);
 
                        this.currentTableName = "Second";

this.listRelations.Add(new ParentTableRelation("Third", "SecondId",TableRelationCardinality.OneToMany,true));
this.listRelations.Add(new ChildTableRelation("First",TableRelationCardinality.OneToOne,"Id","FirstId"));

                  }


			public override DatabaseField[] TableFields 
			{
				get{ return _fields;}
				set{_fields = value;}
			}
          public Second Clone()
          {
                 return this.Clone<Second>();
          }

public System.Int32? Id
{
    get
    {
          return (System.Int32? ) (this.GetField("Id")).fieldValue;
    }

    set
    {
          this.SetFieldValue("Id", value);
    }
}


public System.Int32? FirstId
{
    get
    {
          return (System.Int32? ) (this.GetField("FirstId")).fieldValue;
    }

    set
    {
          this.SetFieldValue("FirstId", value);
    }
}


public System.Int32? Age
{
    get
    {
          return (System.Int32? ) (this.GetField("Age")).fieldValue;
    }

    set
    {
          this.SetFieldValue("Age", value);
    }
}


public System.Int32? Quantity
{
    get
    {
          return (System.Int32? ) (this.GetField("Quantity")).fieldValue;
    }

    set
    {
          this.SetFieldValue("Quantity", value);
    }
}

public Array GetThird()
{
    Type relatedType = Type.GetType("Extender.Third");
    return this.GetRelatedTableData(relatedType);
}
public TableMetadata GetFirst()
{
    Type relatedType = Type.GetType("Extender.First");
    TableMetadata[] result = this.GetRelatedTableData(relatedType);
    return result.Length > 0 ? result[0] : null; 
}
}
}
